In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go
import ipywidgets as widgets

In [4]:
df = pd.read_csv ('ForeignGifts_edu.csv')

df['Clean Gift Received Date'] = pd.to_datetime(df['Foreign Gift Received Date'], unit='D', origin='1899-12-30')

# UVA Findings

In [ ]:
uva_df = df[df['Institution Name'] == 'University of Virginia']

In [7]:
#Count of foreign gifts to UVA
print(uva_df.shape[0])

#Total amount of foreign gifts to UVA
print(uva_df['Foreign Gift Amount'].sum())

98
22189238


The output of the code above shows that UVA has recieved 98 total foreign gifts totaling to $22,189,238.

In [27]:
#Histogram of UVA foreign gift amount
gift_amount_hist = px.histogram(uva_df, x='Foreign Gift Amount', title='Distribution of Amount per Foreign Gift at UVA', labels={'Foreign Gift Amount': 'Foreign Gift Amount ($)'})
gift_amount_hist.update_layout(yaxis_title='Count of Foreign Gifts')

gift_amount_hist.show()

When looking at the distribution of foreign gift amounts to UVA, we see in the graph above that the majority of foreign gifts that UVA recieves are less than $200,000.

In [29]:
#Histogram of country of giftor to UVA
uva_country_hist = px.histogram(uva_df, x='Country of Giftor', 
                                title='Count of Foreign Gifts by Country',
                                labels={'Country of Giftor': 'Country'})
uva_country_hist.update_layout(yaxis_title='Count of Foreign Gifts')

#Bar of gift amount to UVA by country
uva_gift_amount_country = uva_df.groupby('Country of Giftor', as_index=False).sum("Foreign Gift Amount")
uva_gift_amount_bar = px.bar(uva_gift_amount_country, x='Country of Giftor', 
                             y='Foreign Gift Amount', 
                             title='Total Foreign Gift Amount by Country', 
                             labels={'Foreign Gift Amount': 'Total Gift Amount ($)', 'Country of Giftor': 'Country'})

uva_country_hist.show()
uva_gift_amount_bar.show()

Breaking this down by where these gifts are coming from, the graph titled 'Count of Foreign Gifts by Country' shows that UVA recieves the most gifts by far from England. But we can see in the second graph titled 'Total Foreign Gift Amount by Country', that Bangledesh and Sweeden have given UVA the most foreign gifting in dollar amount. These both have a total gifting amount that is at least double that of England.

In [56]:
#Breakdown of number of gifts by gift type
pd.DataFrame(uva_df['Gift Type'].value_counts().reset_index(name = 'Count of Foreign Gifts'))

,Gift Type,Count of Foreign Gifts
0,Contract,64
1,Monetary Gift,34


I next wanted to see what type of foreign gifts UVA was recieving at the highest volume. The tablle above shows that the largest number of foriegn gifts to UVA are contracts and then the only other type of gift recieved is monetary gifts. 

In [58]:
uva_df.groupby("Gift Type").sum("Foreign Gift Amount").sort_values(by="Foreign Gift Amount", ascending=False)[['Foreign Gift Amount']]

,Foreign Gift Amount
Gift Type,
Contract,14807636
Monetary Gift,7381602


When looking at the gift amounts, UVA recieved a higher dollar amount in foreign gifting through contracts at $14,807,636 than through monetary gifting which only earned them $7,381,602

In [59]:
gift_amount_by_type = px.line(uva_df, x='Clean Gift Received Date', y='Foreign Gift Amount', color='Gift Type')
gift_amount_by_type.show()

# Comparison to Other Top Public Universities

In [40]:
#Creating data set for UVA and peer institutions
top_public_universities = ['University of Virginia', 'University of Michigan - Ann Arbor', 'University of North Carolina - Chapel Hill', 'University of California, Berkeley', 'University of California, Los Angeles', 'University of California, San Diego', 'University of Florida', 'University of Texas at Austin']
uva_and_peers_df = df[df['Institution Name'].isin(top_public_universities)]

To be able to put UVA's performance in foreign gifting into context I wanted to compare it to the performance of other top public universities. I chose 7 other universities to compare UVA to: 
- University of Michigan - Ann Arbor
- University of North Carolina - Chapel Hill
- University of California, Berkeley
- University of California, Los Angeles
- University of California, San Diego
- University of Florida, University of Texas at Austin.

In [48]:
#Number of foreign gifts by institution
uva_and_peers_df.groupby("Institution Name").sum("Foreign Gift Amount").sort_values(by="Foreign Gift Amount", ascending=False)[['Foreign Gift Amount']]

,Foreign Gift Amount
Institution Name,
"University of California, Berkeley",294479904
University of Michigan - Ann Arbor,287336783
"University of California, Los Angeles",242146363
"University of California, San Diego",131941755
University of Texas at Austin,85858408
University of North Carolina - Chapel Hill,41292544
University of Florida,24759139
University of Virginia,22189238


First comparing UVA's $22,189,238 total foreign gifting to other the other top public universities I selected, it ranks last. The top of these public universties in terms of total foreign gift amount is Berkely at $2,94,479,904 which is over 13 times that of UVA. University of Michigan - Ann Arbor comes in second at $287,336,783 which is still 12 times that of UVA.

In [54]:
pd.DataFrame(uva_and_peers_df['Institution Name'].value_counts().reset_index(name = 'Count of Foreign Gifts'))

,Institution Name,Count of Foreign Gifts
0,"University of California, Los Angeles",3916
1,University of Michigan - Ann Arbor,715
2,"University of California, Berkeley",394
3,University of Texas at Austin,377
4,University of North Carolina - Chapel Hill,286
5,"University of California, San Diego",203
6,University of Virginia,98
7,University of Florida,33


There is a similar story when looking at the total number of foreign gifts recieved by each institution where UVA is second-to-last followed only by University of Florida. University of California, Los Angeles recieveed the largest number of foreign gifts at 3,916 and University of Michigan - Ann Arbor	came in second at 715.